components/PopUp.js

```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+
+export default function PopUp({ constructionId, handleDelete }) {
+  const [popUp, setPopUp] = useState(false);
+
+  const togglePopUp = () => {
+    setPopUp(!popUp);
+  };
+
+  return (
+    <>
+      <button type="button" onClick={togglePopUp}>
+        Löschen
+      </button>
+
+      {popUp && (
+        <div>
+          <div onClick={togglePopUp}></div>
+          <div>
+            <h2>Möchten Sie die Baustelle wirklich endgültig löschen?</h2>
+            <button type="button" onClick={togglePopUp}>
+              Abbrechen
+            </button>
+            <button type="button" onClick={() => handleDelete(constructionId)}>
+              Löschen
+            </button>
+          </div>
+        </div>
```

Comment:
```suggestion
        <section>
            <h2>Möchten Sie die Baustelle wirklich endgültig löschen?</h2>
            <button type="button" onClick={togglePopUp}>
              Abbrechen
            </button>
            <button type="button" onClick={() => handleDelete(constructionId)}>
              Löschen
            </button>
        </section>
```

I think we can improve the HTML semantic a little bit by removing not necessary `div` elements and this part `<div onClick={togglePopUp}></div>` is not needed anymore or? :)